src/tests/ftest/control/dmg_config_generate.py

```diff
@@ -0,0 +1,155 @@
+"""
+
+from apricot import TestWithServers
+from command_utils import CommandFailure
+
+
+class ConfigGenerate(TestWithServers):
+    """Test Class Description:
+
+    Verify the veracity of the configuration created by the command and what
+    the user specified, input verification and correct execution of the server
+    once the generated configuration is propagated to the servers.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ConfigGenerate object."""
+        super(ConfigGenerate, self).__init__(*args, **kwargs)
+        self._start_servers = False
+
+    def start_server_managers(self):
+        """Start the daos_server processes on each specified list of hosts."""
+        if self._start_servers:
+            super(ConfigGenerate, self).start_server_managers()
+
+    def run_test(self, pmem, nvme, net):
+        """Run test."""
+        self._start_servers = True
+        self.server_managers[-1].manager.job.discover_pmem.value = pmem
+        self.server_managers[-1].manager.job.discover_nvme.value = nvme
+        self.server_managers[-1].manager.job.discover_net.value = net
+
+        self.start_server_managers()
+
+    def test_dmg_config_generate_1(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_1
+        """
+        self.run_test(None, None, None)
+
+    def test_dmg_config_generate_2(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_2
+        """
+        self.run_test(None, 1, None)
+
+    def test_dmg_config_generate_3(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_3
+        """
+        self.run_test(None, 2, None)
+
+    def test_dmg_config_generate_4(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_4
+        """
+        self.run_test(1, None, None)
+
+    def test_dmg_config_generate_5(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
+
+        :avocado: tags=all,small,hw,full_regression,config_generate
+        :avocado: tags=config_generate_4
+        """
+        self.run_test(2, None, None)
```

Comment:
given you are just varying input parameters in the subtests might be sensible to have a single test function that iterates through an array of input values and runs one run_test for each input value set in the array, table driven testing essentially

Reply:
@tanabarr defining multiple `test_dmg_config_generate_*` methods provides some additional ease of use benefits that are more difficult to achieve with an iterative loop or mux'd avocado test variants.  It is easy to isolate a single test variant with the use of tags as opposed to commenting out entries in code or the test yaml.  This also lends itself to easily implementing the `skipForTicket(<JIRA_ticket>)` model if a test ever needs to be skipped for some reason, which provides a much clearer indication that a test is currently skipped rather than commented out test yaml sections.

Reply:
Yes that would be ok as well, the reason I have separate test cases is to keep the functionality for canceling or skipping a particular test case. If I loop through the options the whole test would have to be skipped or canceled if any issues come up.

Reply:
understood thanks for the explanations